gmt/base_plotting.py

```diff
@@ -126,7 +126,48 @@ def coast(self, **kwargs):
             lib.call_module("coast", build_arg_string(kwargs))
 
     @fmt_docstring
-    @use_alias(R="region", J="projection", B="frame", I="shading", C="cmap")
+    @use_alias(
+        R="region",
+        J="projection",
+        A="annotation_interval",
+        C="contour_interval",
+        B="frame",
+        G="label_placement",
```

Comment:
Choosing these names is really hard. I would rename them to single words:
* `A="annotation"`
* `C="interval"` (contour is already in the method name so `fig.grdcontour(contour_interval=..)` seems a bit repetitive)
* `G="label"`

I'm not sure about `limit`. This type of things is often called a `range` in other parts of GMT and some Python functions. But `range` is a builtin function so we should use `ranges` instead. I'm not a huge fan of that. Maybe we can leave `limit` for now until we can think of something better. This is one of those options that should actually be multiple options.

Reply:
Please, feel free to disagree :)  I'm not set on these names and am willing to change

Reply:
You can tell my C bias as this was my attempt at **less verbosity**. 

One thought I had is that `-C` can be aliased as `cmap` as per `countour` since it accepts a `CPT` file  input as well, it just also accepts a number for contour intervals so is not compatible the other way.  @mjziebarth   maybe has some thoughts as well?

Reply:
@leouieda One thing that happens is that something like `contour_interval`  increases the verbosity  when using the functions directly, but makes it clearer to construct keyword dictionaries. Application designers are more likely to use this pattern but it is still useful for managing plot options in scripts.

e.g.

    OPTS={ 
           projection:"W0/6i",
           coast:1, 
           borders:1,
           annotation_interval:1000,
           contour_interval:100,
     }
     fig.coast(**OPTS)
     fig.grdcontour(grd, **OPTS)
     

vs 

    OPTS={ 
           projection:"W0/6i",
           annotation:1000,
           interval:100,
     }
     # ...


More verbose names are easier to work with, and additionally reusing the keywords for incompatible  parameters  can prevent user from using a single dictionary to store common keyword arguments, but  on the other hand this is already true in some cases and is certainly true for many GMT single letter arguments...

In any case I will take your advice for the names,  and instead put my thoughts in #219 because I think this is a bigger discussion than `grdcontour`

Reply:
From someone new to GMT: The more verbose names may make the difference between intuitively being able to write the code and having to consult the man pages.
But I agree with @akshmakov that this is probably best for general discussion e.g. in #219

Reply:
Yeah, I agree. All of this will have to be updated at some point. For now, we need some time to use these functions and see where they fall short. No point in trying to predict all possible use cases.

Reply:
Updated with the following simplified keywords

```python
A="annotation",
B="frame",
C="interval",
G="label_placement",
J="projection",
L="limit",
Q="cut",
R="region",
S="resample",
U="logo",
W="pen",
```